### PR TITLE
Feature/drawer nav item hidden

### LINKED ISSUE #185: Hidden Nav Items
#### Describe the desired behavior
User should be able to conditionally hide nav items that are defined in the items list.

#### Describe the current behavior
All items are rendered. No option to hide them.

#### Is this request related to a current issue?
Adopter Request.

#### Additional Context
The primary use case for this is role-based routes. Users should not be able to see/access some of the admin only pages of the application. There is an equivalent issue already for React and React Native.
https://github.com/pxblue/react-component-library/issues/197
https://github.com/pxblue/react-native-component-library/issues/83

### PR DESCRIPTION
**Pull request by emclaug2**
Fixes #185 

#### Changes proposed in this Pull Request:
- Add `isVisible` prop to nav items.  Default value is `true`.

### CODE REVIEW

**Comment by huayunh** [comment]
I think "isHidden" or just "hidden" might be more intuitive than "isVisible"

I assume that everytime you introduce a new prop/attribute, it is a derivative from the default use case. 

Since the default is visible, if the user wants to modify it from the default, they will use the"hidden" attribute

**Comment by emclaug2** [comment]
> I think "isHidden" or just "hidden" might be more intuitive than "isVisible"
> 
> I assume that everytime you introduce a new prop/attribute, it is a derivative from the default use case.
> 
> Since the default is visible, if the user wants to modify it from the default, they will use the"hidden" attribute

I like this approach; I've updated the prop name to `isHidden`.

File: components/src/core/drawer/drawer-body/nav-item/drawer-nav-item.component.ts [hunk 1]
```diff
@@ -117,6 +120,7 @@ export class DrawerNavItemComponent extends StateListener implements Omit<Drawer
     @Input() statusColor: string;
     @Input() subtitle: string;
     @Input() title: string;
+    @Input() isHidden = false;
```
Comment (joebochill) [inline components/src/core/drawer/drawer-body/nav-item/drawer-nav-item.component.ts:123]: correct me if I'm wrong, but I don't think we typically use the `is_` prefix for attributes/props/Inputs. I would go for `hidden` instead if it were me.

File: docs/Drawer.md [hunk 2]
```diff
@@ -304,6 +304,7 @@ Parent element (`<pxb-drawer-nav-item>`) attributes:
 | divider                   | Sets whether to show divider          | `boolean`            | no       | true    |
 | expanded                  | Sets whether to show nested nav items | `boolean`            | no       | false   |
 | hidePadding               | Sets whether to show/hide padding     | `boolean`            | no       |         |
+| isHidden                  | Sets whether to show the nav item     | `boolean`            | no       | false   |
```
Comment (joebochill) [inline docs/Drawer.md:307]:
```suggestion
| hidden                  | Sets whether to hide the nav item     | `boolean`            | no       | false   |
```

**Review by joebochill — APPROVED** [review]

**Comment by joebochill** [comment]
Unit test to verify that it works?

**Comment by joebochill** [comment]
Anything we want to show in Storybook?

**Comment by emclaug2** [comment]
> Unit test to verify that it works?

I'll add a unit test to show that it works; I don't think this change is big enough to warrant it's own story in storybook.  I've considered adding a knob to the full config story but found a new knob a bit confusing.    I think the api docs should be enough for people to know how to use this new prop.

File: components/src/core/drawer/drawer-body/nav-item/drawer-nav-item.component.ts (hunk 1 above)
Comment (emclaug2) [reply components/src/core/drawer/drawer-body/nav-item/drawer-nav-item.component.ts:123]: I've removed the `is` part of the `isHidden`

**Pull request merged**